pkg/apis/integreatly/v1alpha1/installation_types.go

```diff
@@ -91,6 +92,17 @@ type InstallationSpec struct {
 	NamespacePrefix  string         `json:"namespacePrefix"`
 	SelfSignedCerts  bool           `json:"selfSignedCerts"`
 	PullSecret       PullSecretSpec `json:"pullSecret,omitempty"`
+
+	// SMTPSecret is the name of a secret in the installation
+	// namespace containing SMTP connection details. The secret
+	// must contain the following fields:
+	//
+	// host
+	// port
+	// tls
+	// username
+	// password
+	SMTPSecret string `json:"smtpSecret,omitempty"`
```

Comment:
Should I change this to be required (remove the `,omitempty`)?

Reply:
I think it's fine to leave it with `,omitempty` for now

Reply:
One of the concerns with making these new fields required is that it makes the CRD backwards-incompatible with the previous release, and people wouldn't be able to upgrade without manual intervention.

I think the way that this would generally work is creating a new resource version (say, v1alpha2). That would be a different can of worms that I'd rather not have to deal with if possible! :sweat_smile:

Reply:
Well. then it seems like we've a valid reason to leave it as it is :smile: